### PR TITLE
fix several issues reported by cppcheck

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fix several issues reported by cppcheck.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: -
  - [ ] Backport for 3.9: -
  - [ ] Backport for 3.8: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/GeneralServer/CommTask.cpp [hunk 1]
```diff
@@ -447,21 +447,18 @@ void CommTask::executeRequest(std::unique_ptr<GeneralRequest> request,
   TRI_ASSERT(request != nullptr);
   TRI_ASSERT(response != nullptr);
 
+  if (request == nullptr || response == nullptr) {
```
Comment (MBkkt) [inline arangod/GeneralServer/CommTask.cpp:450]: We have asserts

**Review by MBkkt — APPROVED** [review]

File: arangod/GeneralServer/CommTask.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/GeneralServer/CommTask.cpp:450]: Yes, but the asserts will not fire in production code. That's why I also added the exception.
The previous code contained some `if (request) ... else if (response) ...`, also these things were present only after dereferencing the pointers :+1:

**Comment by jsteemann** [comment]
cppcheck run:
https://jenkins01.arangodb.biz/job/arangodb-ANY-cppcheck/170/

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**